### PR TITLE
[MeshModel] Add new relationship definitions

### PR DESCRIPTION
**Pull request by Boombag0607**
**Notes for Reviewers**

This PR adds Controller and Configurational peer relationship definitions.

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit ccaae662d83e878dc5fb4a3e0e7b515be42c1103 at: https://64bfa57719325c520c6195c4--meshery-docs-preview.netlify.app

File: server/meshmodel/relationships/peer_configurational_relationship.json [hunk 1]
```diff
@@ -0,0 +1,65 @@
+{
```
Comment (leecalcote) [inline server/meshmodel/relationships/peer_configurational_relationship.json:1]: @MUzairS15, please confirm whether these new relationship definitions will be automatically accounted for on Meshery Server boot. If they will, then @Boombag0607, further scrutiny is required. Define manual tests and show the outputs of your results; e.g.

❌ A ConfigMap of one Kubernetes version (e.g. v1.25.0) can relate to a Service within a different Kubernetes version (e.g. v1.27.1).
✅ A ConfigMap of one Kubernetes version (e.g. v1.25.0) can relate to a Service within a different Kubernetes version (e.g. v1.25.0).

**Review by leecalcote — CHANGES_REQUESTED** [review]
Let's define and agree upon the definition of a "Configurational" type and a kind "Peer". Neither of these two relationship kinds/subtypes are currently defined in MeshModal.

File: server/meshmodel/relationships/peer_configurational_relationship.json (hunk 1 above)
Comment (Boombag0607) [reply server/meshmodel/relationships/peer_configurational_relationship.json:1]: @leecalcote I'll take a look if these definitions will sit well. Or we can focus on getting the MeshModal relationship doc approved first, then code them in the server.

**Comment by Chadha93** [comment]
@Boombag0607  Let's discuss this on the Meshery Dev call. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1Dg4Uz9rGTuGHmYZwIt47z1j5x3LmnpxD1M1SgypAkQ0/edit#) if you would. :)

**Comment by leecalcote** [comment]
merge conflict

**Comment by Boombag0607** [comment]
Closing this.

**Pull request closed without merging**